src/binary-reader-ir.cc

```diff
@@ -868,36 +865,36 @@ Result BinaryReaderIR::OnLocalNameLocalCount(Index index, Index count) {
 }
 
 Result BinaryReaderIR::OnInitExprF32ConstExpr(Index index, uint32_t value) {
-  Expr* expr = Expr::CreateConst(Const(Const::F32(), value, GetLocation()));
+  auto expr = new ConstExpr(Const(Const::F32(), value, GetLocation()));
```

Comment:
Maybe I need to get over it but I'm used to expecting to see fewer "new" calls in modern C++.  Could/Should we use movable types here,  or emplace, or make_unique?